06_HelloGraphicsQueue/main.cpp

```diff
@@ -47,6 +47,32 @@ class HelloGraphicsQueueApp final : public examples::MonoDeviceApplication, publ
 			// - save image to disk
 			// all without using IUtilities
 
+			auto redImage = core::make_smart_refctd_ptr< IGPUImage >;
+			auto blueImage = core::make_smart_refctd_ptr< IGPUImage >;
+
+			IGPUImageView::SCreationParams redParams;
+			redParams.flags = static_cast<IGPUImageView::E_CREATE_FLAGS>(0u);
+			redParams.format = EF_R8G8B8A8_SRGB;
+			redParams.image = core::smart_refctd_ptr<ICPUImage>(redImage);
+			redParams.viewType = IGPUImageView::ET_2D;
+			redParams.subresourceRange = { static_cast<IImage::E_ASPECT_FLAGS>(0u),0u,1u,0u,1u };
+			smart_refctd_ptr<IGPUImageView> redImageView = core::make_smart_refctd_ptr<IGPUImageView>(m_device, redParams);
+
+			IGPUImageView::SCreationParams blueParams;
+			blueParams.flags = static_cast<IGPUImageView::E_CREATE_FLAGS>(0u);
+			blueParams.format = EF_R8G8B8A8_SRGB;
+			blueParams.image = core::smart_refctd_ptr<IGPUImage>(blueImage);
+			blueParams.viewType = IGPUImageView::ET_2D;
+			blueParams.subresourceRange = { static_cast<IImage::E_ASPECT_FLAGS>(0u),0u,1u,0u,1u };
+			smart_refctd_ptr<IGPUImageView> redImageView = core::make_smart_refctd_ptr<IGPUImageView>(m_device, blueParams);
+
+
+			auto buff = core::make_smart_refctd_ptr<nbl::asset::IBuffer>;
```

Comment:
This would need to be after the bufferParams, and you'd have the params in the constructor:
```cpp
			auto buff = core::make_smart_refctd_ptr<nbl::asset::IBuffer>(bufferParams);
```

Reply:
IMHO because Nahim is supposed to fill the images using Graphics Queue commands, there's no need for a buffer except when downloading the image (which happens after the dispatch